### ISSUE TITLE
Running STFService on user devices

### ISSUE THREAD

**Issue by paulirish** [openstf/STFService.apk#1]
On the way to report this I see the "not meant for user devices" text… :)

For me, the benefits of using openstf on a small set of devices are huge. Two of those devices are my personal phones. (yeah, two in my pocket all the time. shrug.)

In this case, the stfservice continues to run long after i've disconnected.

I don't know what the best solution should be, but I would like to easily use STP on and off with my own devices and have the STFService not always running. Basically I think I'm asking for an easier off-switch than Settings > Apps > Running > STF > Terminate service.

**Comment by sorccu** [comment]
You're right, I've been thinking about the same thing. All of our devices are fixed right now, so it hasn't been a real issue for us as they're going to be running it anyway, but I'm aware of the problem.

Just to be clear: you'd be happ(y|ier) if it was easier to turn the service off, or would you prefer if the service stopped itself when STF stops?

**Comment by paulirish** [comment]
1st preference: the android service terminated when STF stops. 
2nd: manually kill the service from the notification bar.

**Comment by sorccu** [comment]
The service opens a socket server on the device which STF then connects to. My current war plan is to make the service terminate itself if there's not even a single connection for ~30s or so. This should achieve the desired result albeit with a small delay, and as a bonus would be relatively easy to implement.

The `UsbManager` broadcasts are for SDK 12+ and I'm not completely sure if they'd work for this use case anyway, although I'll probably do some additional googling later.

**Comment by paulirish** [comment]
>  My current war plan is to make the service terminate itself if there's not even a single connection for ~30s or so

works for me! thanks

**Comment by alisamir82** [comment]
There is a plan to add more to the events list under PhoneStateListener, I tried to add 	LISTEN_CELL_INFO ( to report cell data changes) or LISTEN_SIGNAL_STRENGTHS, both didn't work, I managed to build the apk with no issues but nothing got reported. ! I thought I would ask here it is a known issue. thanks.

**Comment by deano2390** [comment]
+1
